File: ce/customer-engagement/on-premises/snippets/csharp/CRMV8/scheduleandappointment/cs/queryworkinghoursofuser1.cs
// <snippetqueryworkinghoursofuser1>


 // Get the current user's information.
 WhoAmIRequest userRequest = new WhoAmIRequest();
 WhoAmIResponse userResponse = (WhoAmIResponse)_serviceProxy.Execute(userRequest);

 // Retrieve the working hours of the current user.                                              
 QueryScheduleRequest scheduleRequest = new QueryScheduleRequest
 {
     ResourceId = userResponse.UserId,
     Start = DateTime.Now,
     End = DateTime.Today.AddDays(7),
     TimeCodes = new TimeCode[] { TimeCode.Available }
 };
 QueryScheduleResponse scheduleResponse = (QueryScheduleResponse)_serviceProxy.Execute(scheduleRequest);

 // Verify if some data is returned for the availability of the current user
 if (scheduleResponse.TimeInfos.Length > 0)
 {
     Console.WriteLine("Successfully queried the working hours of the current user.");
 }

// </snippetqueryworkinghoursofuser1>

File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/scheduleandappointmentvb/vb/queryworkinghoursofuser1.vb
' <snippetqueryworkinghoursofuser1>


 ' Get the current user's information.
 Dim userRequest As New WhoAmIRequest()
 Dim userResponse As WhoAmIResponse = CType(_serviceProxy.Execute(userRequest), WhoAmIResponse)

 ' Retrieve the working hours of the current user.                                              
 Dim scheduleRequest As QueryScheduleRequest = New QueryScheduleRequest With { _
     .ResourceId = userResponse.UserId, .Start = Date.Now, .End = Date.Today.AddDays(7), _
     .TimeCodes = New TimeCode() {TimeCode.Available}}
 Dim scheduleResponse As QueryScheduleResponse = CType(_serviceProxy.Execute(scheduleRequest), QueryScheduleResponse)

 ' Verify if some data is returned for the availability of the current user
 If scheduleResponse.TimeInfos.Length > 0 Then
     Console.WriteLine("Successfully queried the working hours of the current user.")
 End If

' </snippetqueryworkinghoursofuser1>